src/oem-debugging/index.md

```diff
@@ -0,0 +1,359 @@
+
+# Debugging Dart code
+
+Use your IDE for standard Dart debugging. These instructions describe Android
+Studio, but you can use your preferred IDE with the Flutter and Dart
+plugins installed and configured.
+
+<aside class="alert alert-success" markdown="1">
+<i class="fa fa-lightbulb-o"> </i> **Tip:**
+Connect to a physical device when debugging, rather than an emulator or
+simulator, which don't support profile mode. For more information, see
+[Flutter's modes](https://github.com/flutter/flutter/wiki/Flutter's-modes).
+</aside>
+
+## Dart debugger
+
+* Open your project in Android Studio. If you don't have a project yet,
+  create one using the instructions in [Test drive](/get-started/test-drive).
+
+* Simultaneously bring up the Debug pane and run the app in the Console
+  view by clicking the bug icon.
+  <img src="images/debug-run.png" style="border: 1px solid #ccc;"
+   alt="looks like a bug superimposed with a tiny green run triangle">
+
+  The first time you launch the app is the slowest.
+  You should see the Debug pane appear at the bottom of the window that
+  looks something like the following:
+
+  <center><img src="images/debug-pane.png" style="border: 1px solid #ccc;"
+   alt="the debug pane showing that it's connected to the debugger and running the app"></center>
+
+  You can configure where the debug pane appears, or even tear it off to its own
+  window using the gear to the right in the Debug pane bar.
+  This is true for any inspector in Android Studio.
+
+* Add a breakpoint on the `counter++` line.
+
+* Hot reload the app.
+  <img src="images/hot-reload-button.png" style="border: 1px solid #ccc;"
+   alt="looks like a lightning bolt">
+
+* In the app, click the **+** button (FloatingActionButton, or FAB, for short)
+  to increment the counter. The app pauses.
+
+* The following screenshot shows:
+  * Breakpoint in the edit pane.
+  * State of the app in the debug pane, when paused at the breakpoint.
+  * `this` variable expanded to display its values.
+
+  <center><img src="images/debug-pane-action.png" style="border: 1px solid #ccc;"
+   alt="the status of the app when hitting the set breakpoint"></center>
+
+You can step in, out, and over Dart statements, hot reload or resume the app,
+and use the debugger in the same way you'd use any debugger.
+The **5: Debug** button toggles display of the debug pane.
+
+## Flutter inspector
+
+There are two other features provided by the Flutter plugin that you may
+find useful.  The Flutter inspector is a tool for visualizing and exploring
+the Flutter widget tree and helps you:
+
+* Understand existing layouts
+* Diagnose layout issues
+
+Toggle display of the inspector using the vertical button to the
+right of the Android Studio window.
+
+  <center><img src="images/flutter-inspector.png" style="border: 1px solid #ccc;"
+   alt="screenshot showing the Flutter inspector"></center>
+
+## Flutter outline
+
+The Flutter Outline displays the current state of the widget tree.
```

Comment:
It doesn't display the current state of the widget tree. It displays a tree visualization of the build method which may or may not be the same as the widget tree for the build method .